### ISSUE TITLE
Test cases failed

### ISSUE THREAD

**Issue by jacktang** [OpenZeppelin/zeppelin-solidity#770]
## 🎉 Description

I clone the clean source code and install deps via `npm install` and run the tests using `npm run test` and it failed. (logs are posted below) How can I run the test cases smoothly?

## 💻 Environment

* node version: v8.4.0
* npm version: 5.6.0

## 📝 Detail logs

```
22:44 $ npm run test

> zeppelin-solidity@1.7.0 test /Users/jacktang/Code/zeppelin-solidity
> scripts/test.sh

Starting our own testrpc instance
Compiling ./contracts/Bounty.sol...
...
Compiling ./contracts/token/ERC827/ERC827Token.sol...
Compiling ./test/mocks/MathMock.sol...

Compilation warnings encountered:

/Users/jacktang/Code/zeppelin-solidity/contracts/crowdsale/Crowdsale.sol:112:34: Warning: Unused function parameter. Remove or comment out the variable name to silence this warning.
  function _postValidatePurchase(address _beneficiary, uint256 _weiAmount) internal {
                                 ^------------------^
...
Spanning multiple lines.
,/Users/jacktang/Code/zeppelin-solidity/contracts/ownership/HasNoTokens.sol:21:3: Warning: Function state mutability can be restricted to pure
  function tokenFallback(address from_, uint256 value_, bytes data_) external {
  ^
Spanning multiple lines.



  Contract: Bounty
    ✓ sets reward (1046ms)
    ✓ empties itself when destroyed (1627ms)
    Against secure contract
      ✓ cannot claim reward (2780ms)
    Against broken contract
      1) claims reward

    Events emitted during test:
    ---------------------------

    TargetCreated(createdAddress: 0x89c4d9ad6fcd9dd90a1ca1198e15ea5701cbc812)

    ---------------------------

  Contract: DayLimit
/Users/jacktang/Code/zeppelin-solidity/node_modules/ethereumjs-testrpc/build/cli.node.js:68497
var Module;if(!Module)Module=(typeof Module!=="undefined"?Module:null)||{};var moduleOverrides={};for(var key in Module){if(Module.hasOwnProperty(key)){moduleOverrides[key]=Module[key]}}var ENVIRONMENT_IS_WEB=false;var ENVIRONMENT_IS_WORKER=false;var ENVIRONMENT_IS_NODE=false;var ENVIRONMENT_IS_SHELL=false;if(Module["ENVIRONMENT"]){if(Module["ENVIRONMENT"]==="WEB"){ENVIRONMENT_IS_WEB=true}else if(Module["ENVIRONMENT"]==="WORKER"){ENVIRONMENT_IS_WORKER=true}else if(Module["ENVIRONMENT"]==="NODE"){ENVIRONMENT_IS_NODE=true}else if(Module["ENVIRONMENT"]==="SHELL"){ENVIRONMENT_IS_SHELL=true}else{throw new Error("The provided Module['ENVIRONMENT'] value is not valid. It must be one of: WEB|WORKER|NODE|SHELL.")}}else{ENVIRONMENT_IS_WEB=typeof window==="object";ENVIRONMENT_IS_WORKER=typeof importScripts==="function";ENVIRONMENT_IS_NODE=typeof process==="object"&&"function"==="function"&&!ENVIRONMENT_IS_WEB&&!ENVIRONMENT_I

TypeError: Cannot read property 'toJSON' of null
    at /Users/jacktang/Code/zeppelin-solidity/node_modules/ethereumjs-testrpc/build/cli.node.js:85028:24
    at /Users/jacktang/Code/zeppelin-solidity/node_modules/ethereumjs-testrpc/build/cli.node.js:21795:5
    at /Users/jacktang/Code/zeppelin-solidity/node_modules/ethereumjs-testrpc/build/cli.node.js:82948:7
    at /Users/jacktang/Code/zeppelin-solidity/node_modules/ethereumjs-testrpc/build/cli.node.js:7542:16
    at /Users/jacktang/Code/zeppelin-solidity/node_modules/ethereumjs-testrpc/build/cli.node.js:7759:32
    at /Users/jacktang/Code/zeppelin-solidity/node_modules/ethereumjs-testrpc/build/cli.node.js:7534:16
    at /Users/jacktang/Code/zeppelin-solidity/node_modules/ethereumjs-testrpc/build/cli.node.js:82943:9
    at Object.decode (/Users/jacktang/Code/zeppelin-solidity/node_modules/ethereumjs-testrpc/build/cli.node.js:43920:5)
    at /Users/jacktang/Code/zeppelin-solidity/node_modules/ethereumjs-testrpc/build/cli.node.js:82940:20
    at /Users/jacktang/Code/zeppelin-solidity/node_modules/ethereumjs-testrpc/build/cli.node.js:7671:20
    2) "before all" hook: prepare suite

  Contract: Heritable
    3) "before all" hook: prepare suite

  Contract: LimitBalance
    4) "before all" hook: prepare suite

  Contract: ReentrancyGuard
    5) "before all" hook: prepare suite

  Contract: SimpleSavingsWallet
    6) "before all" hook: prepare suite

  Contract: AllowanceCrowdsale
    7) "before all" hook: prepare suite

  Contract: CappedCrowdsale
    8) "before all" hook: prepare suite

  Contract: Crowdsale
    9) "before all" hook: prepare suite

  Contract: FinalizableCrowdsale
    10) "before all" hook: prepare suite

  Contract: IncreasingPriceCrowdsale
    11) "before all" hook: prepare suite

  Contract: IndividuallyCappedCrowdsale
    12) "before all" hook: prepare suite

  Contract: MintedCrowdsale
    13) "before all" hook: prepare suite


npm ERR! code ELIFECYCLE
npm ERR! errno 50
npm ERR! zeppelin-solidity@1.7.0 test: `scripts/test.sh`
npm ERR! Exit status 50
npm ERR!
npm ERR! Failed at the zeppelin-solidity@1.7.0 test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

**Comment by jacktang** [comment]
I also tried to run `test/crowdsale/Crowdsale.test.js` single test. The command `truffle test test/crowdsale/Crowdsale.test.js --network testrpc` and each account is filled 1000000000000000000000000eth as test.sh shows. Some cases failed again :(
```
  Contract: Crowdsale
    accepting payments
      ✓ should accept payments (192ms)
    high-level purchase
      ✓ should log purchase (150ms)
      ✓ should assign tokens to sender (99ms)
      1) should forward funds to wallet

    Events emitted during test:
    ---------------------------

    Transfer(from: <indexed>, to: <indexed>, value: 1e+22)
    Transfer(from: <indexed>, to: <indexed>, value: 1e+22)

    ---------------------------
    low-level purchase
      ✓ should log purchase (119ms)
      2) should assign tokens to beneficiary

    Events emitted during test:
    ---------------------------

    Transfer(from: <indexed>, to: <indexed>, value: 1e+22)
    Transfer(from: <indexed>, to: <indexed>, value: 1e+22)

    ---------------------------
      3) should forward funds to wallet

    Events emitted during test:
    ---------------------------

    Transfer(from: <indexed>, to: <indexed>, value: 1e+22)
    Transfer(from: <indexed>, to: <indexed>, value: 1e+22)

    ---------------------------


  4 passing (4s)
  3 failing

  1) Contract: Crowdsale high-level purchase should forward funds to wallet:
     Error: Error: sender doesn't have enough funds to send tx. The upfront cost is: 42672197500000000000 and the sender's account only has: 15983503000000000000
    at runCall (/Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/ganache-cli/build/cli.node.js:76506:10)
    at /Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/ganache-cli/build/cli.node.js:13430:24
    at replenish (/Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/ganache-cli/build/cli.node.js:10562:17)
    at iterateeCallback (/Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/ganache-cli/build/cli.node.js:10547:17)
    at /Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/ganache-cli/build/cli.node.js:10522:16
    at /Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/ganache-cli/build/cli.node.js:13435:13
    at /Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/ganache-cli/build/cli.node.js:72718:16
    at replenish (/Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/ganache-cli/build/cli.node.js:72665:25)
    at /Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/ganache-cli/build/cli.node.js:72674:9
    at eachLimit (/Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/ganache-cli/build/cli.node.js:72598:36)
      at Object.InvalidResponse (/Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/truffle/build/webpack:/~/web3/lib/web3/errors.js:38:1)
      at /Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/truffle/build/webpack:/~/web3/lib/web3/requestmanager.js:86:1
      at /Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/truffle/build/webpack:/~/truffle-provider/wrapper.js:134:1
      at XMLHttpRequest.request.onreadystatechange (/Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/truffle/build/webpack:/~/web3/lib/web3/httpprovider.js:128:1)
      at XMLHttpRequestEventTarget.dispatchEvent (/Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/truffle/build/webpack:/~/xhr2/lib/xhr2.js:64:1)
      at XMLHttpRequest._setReadyState (/Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/truffle/build/webpack:/~/xhr2/lib/xhr2.js:354:1)
      at XMLHttpRequest._onHttpResponseEnd (/Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/truffle/build/webpack:/~/xhr2/lib/xhr2.js:509:1)
      at IncomingMessage.<anonymous> (/Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/truffle/build/webpack:/~/xhr2/lib/xhr2.js:469:1)

  2) Contract: Crowdsale low-level purchase should assign tokens to beneficiary:
     Error: Error: sender doesn't have enough funds to send tx. The upfront cost is: 42672197500000000000 and the sender's account only has: 15984637200000000000
    at runCall (/Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/ganache-cli/build/cli.node.js:76506:10)
    at /Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/ganache-cli/build/cli.node.js:13430:24
    at replenish (/Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/ganache-cli/build/cli.node.js:10562:17)
    at iterateeCallback (/Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/ganache-cli/build/cli.node.js:10547:17)
    at /Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/ganache-cli/build/cli.node.js:10522:16
    at /Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/ganache-cli/build/cli.node.js:13435:13
    at /Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/ganache-cli/build/cli.node.js:72718:16
    at replenish (/Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/ganache-cli/build/cli.node.js:72665:25)
    at /Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/ganache-cli/build/cli.node.js:72674:9
    at eachLimit (/Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/ganache-cli/build/cli.node.js:72598:36)
      at Object.InvalidResponse (/Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/truffle/build/webpack:/~/web3/lib/web3/errors.js:38:1)
      at /Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/truffle/build/webpack:/~/web3/lib/web3/requestmanager.js:86:1
      at /Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/truffle/build/webpack:/~/truffle-provider/wrapper.js:134:1
      at XMLHttpRequest.request.onreadystatechange (/Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/truffle/build/webpack:/~/web3/lib/web3/httpprovider.js:128:1)
      at XMLHttpRequestEventTarget.dispatchEvent (/Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/truffle/build/webpack:/~/xhr2/lib/xhr2.js:64:1)
      at XMLHttpRequest._setReadyState (/Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/truffle/build/webpack:/~/xhr2/lib/xhr2.js:354:1)
      at XMLHttpRequest._onHttpResponseEnd (/Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/truffle/build/webpack:/~/xhr2/lib/xhr2.js:509:1)
      at IncomingMessage.<anonymous> (/Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/truffle/build/webpack:/~/xhr2/lib/xhr2.js:469:1)

  3) Contract: Crowdsale low-level purchase should forward funds to wallet:
     Error: Error: sender doesn't have enough funds to send tx. The upfront cost is: 42672197500000000000 and the sender's account only has: 15984637200000000000
    at runCall (/Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/ganache-cli/build/cli.node.js:76506:10)
    at /Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/ganache-cli/build/cli.node.js:13430:24
    at replenish (/Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/ganache-cli/build/cli.node.js:10562:17)
    at iterateeCallback (/Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/ganache-cli/build/cli.node.js:10547:17)
    at /Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/ganache-cli/build/cli.node.js:10522:16
    at /Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/ganache-cli/build/cli.node.js:13435:13
    at /Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/ganache-cli/build/cli.node.js:72718:16
    at replenish (/Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/ganache-cli/build/cli.node.js:72665:25)
    at /Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/ganache-cli/build/cli.node.js:72674:9
    at eachLimit (/Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/ganache-cli/build/cli.node.js:72598:36)
      at Object.InvalidResponse (/Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/truffle/build/webpack:/~/web3/lib/web3/errors.js:38:1)
      at /Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/truffle/build/webpack:/~/web3/lib/web3/requestmanager.js:86:1
      at /Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/truffle/build/webpack:/~/truffle-provider/wrapper.js:134:1
      at XMLHttpRequest.request.onreadystatechange (/Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/truffle/build/webpack:/~/web3/lib/web3/httpprovider.js:128:1)
      at XMLHttpRequestEventTarget.dispatchEvent (/Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/truffle/build/webpack:/~/xhr2/lib/xhr2.js:64:1)
      at XMLHttpRequest._setReadyState (/Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/truffle/build/webpack:/~/xhr2/lib/xhr2.js:354:1)
      at XMLHttpRequest._onHttpResponseEnd (/Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/truffle/build/webpack:/~/xhr2/lib/xhr2.js:509:1)
      at IncomingMessage.<anonymous> (/Users/jacktang/.nvm/versions/node/v8.4.0/lib/node_modules/truffle/build/webpack:/~/xhr2/lib/xhr2.js:469:1)
```

**Comment by jacktang** [comment]
I found the failed reason. I change `development` port to the ganache server. And today I rollback truffle config, and run the test.js, still some errors
```
  372 passing (2m)
  2 failing

  1) Contract: IndividuallyCappedCrowdsale individual capping accepting payments should reject payments outside cap:
     Error: VM Exception while processing transaction: revert
      at Object.InvalidResponse (node_modules/truffle/build/cli.bundled.js:43303:16)
      at node_modules/truffle/build/cli.bundled.js:331156:36
      at node_modules/truffle/build/cli.bundled.js:314196:9
      at XMLHttpRequest.request.onreadystatechange (node_modules/truffle/build/cli.bundled.js:315621:13)
      at XMLHttpRequestEventTarget.dispatchEvent (node_modules/truffle/build/cli.bundled.js:70159:18)
      at XMLHttpRequest._setReadyState (node_modules/truffle/build/cli.bundled.js:70449:12)
      at XMLHttpRequest._onHttpResponseEnd (node_modules/truffle/build/cli.bundled.js:70604:12)
      at IncomingMessage.<anonymous> (node_modules/truffle/build/cli.bundled.js:70564:24)

  2) Contract: IndividuallyCappedCrowdsale group capping reporting state should report correct cap:

      AssertionError: expected '0' to equal '2000000000000000000'
      + expected - actual

      -0
      +2000000000000000000

      at Context._callee13$ (test/crowdsale/IndividuallyCappedCrowdsale.test.js:101:45)
      at tryCatch (node_modules/regenerator-runtime/runtime.js:65:40)
      at Generator.invoke [as _invoke] (node_modules/regenerator-runtime/runtime.js:303:22)
      at Generator.prototype.(anonymous function) [as next] (node_modules/regenerator-runtime/runtime.js:117:21)
      at step (test/crowdsale/IndividuallyCappedCrowdsale.test.js:15:191)
      at test/crowdsale/IndividuallyCappedCrowdsale.test.js:15:361
      at <anonymous>
      at process._tickCallback (internal/process/next_tick.js:188:7)
```
And it is very confuse to me that CappedCrowdsale sometime failed with error and sometime passed (running single testcase is ok)
```
  1) Contract: CappedCrowdsale accepting payments should reject payments outside cap:
     Error: VM Exception while processing transaction: revert
      at Object.InvalidResponse (node_modules/truffle/build/cli.bundled.js:43303:16)
      at node_modules/truffle/build/cli.bundled.js:331156:36
      at node_modules/truffle/build/cli.bundled.js:314196:9
      at XMLHttpRequest.request.onreadystatechange (node_modules/truffle/build/cli.bundled.js:315621:13)
      at XMLHttpRequestEventTarget.dispatchEvent (node_modules/truffle/build/cli.bundled.js:70159:18)
      at XMLHttpRequest._setReadyState (node_modules/truffle/build/cli.bundled.js:70449:12)
      at XMLHttpRequest._onHttpResponseEnd (node_modules/truffle/build/cli.bundled.js:70604:12)
      at IncomingMessage.<anonymous> (node_modules/truffle/build/cli.bundled.js:70564:24)
```

I also tried to modify Line64 in test.js to `node_modules/.bin/truffle test "$@"  --network testrpc` but it didn't work :(

**Comment by jacktang** [comment]
IndividuallyCappedCrowdsale still can't pass the testing, see below
```
  373 passing (2m)
  1 failing

  1) Contract: IndividuallyCappedCrowdsale group capping accepting payments should reject payments outside cap:
     Error: VM Exception while processing transaction: revert
      at Object.InvalidResponse (node_modules/truffle/build/cli.bundled.js:43303:16)
      at node_modules/truffle/build/cli.bundled.js:331156:36
      at node_modules/truffle/build/cli.bundled.js:314196:9
      at XMLHttpRequest.request.onreadystatechange (node_modules/truffle/build/cli.bundled.js:315621:13)
      at XMLHttpRequestEventTarget.dispatchEvent (node_modules/truffle/build/cli.bundled.js:70159:18)
      at XMLHttpRequest._setReadyState (node_modules/truffle/build/cli.bundled.js:70449:12)
      at XMLHttpRequest._onHttpResponseEnd (node_modules/truffle/build/cli.bundled.js:70604:12)
      at IncomingMessage.<anonymous> (node_modules/truffle/build/cli.bundled.js:70564:24)
```

**Comment by fiiiu** [comment]
Thanks @jacktang! We missed that file, fixed in #789.

**Comment by jacktang** [comment]
I tried to run the `script/test.sh` four times, and sometime the tests failed at
```
  364 passing (3m)
  10 failing

  1) Contract: Bounty Against secure contract cannot claim reward:
     AssertionError: expected '1000000000000000000' to equal 2000000000000000000
      at _callee3$ (test/Bounty.test.js:59:16)
      at tryCatch (node_modules/regenerator-runtime/runtime.js:65:40)
      at Generator.invoke [as _invoke] (node_modules/regenerator-runtime/runtime.js:303:22)
      at Generator.prototype.(anonymous function) [as next] (node_modules/regenerator-runtime/runtime.js:117:21)
      at step (test/Bounty.test.js:3:191)
      at test/Bounty.test.js:3:437
      at Promise (<anonymous>)
      at test/Bounty.test.js:3:99
      at watcher (test/Bounty.test.js:45:31)
      at wrappedHandler (test/Bounty.test.js:15:23)

  2) Contract: MintedCrowdsale accepting payments should be token owner:
     Error: Invalid number of arguments to Solidity function
```
or failed at

```
  365 passing (3m)
  9 failing

  1) Contract: MintedCrowdsale accepting payments should be token owner:
     Error: Invalid number of arguments to Solidity function
      at Object.InvalidNumberOfSolidityArgs (node_modules/truffle/build/cli.bundled.js:43290:16)
      at SolidityFunction.validateArgs (node_modules/truffle/build/cli.bundled.js:329518:22)
      at SolidityFunction.toPayload (node_modules/truffle/build/cli.bundled.js:329534:10)
      at SolidityFunction.call (node_modules/truffle/build/cli.bundled.js:329575:24)
      at SolidityFunction.execute (node_modules/truffle/build/cli.bundled.js:329704:22)
      at node_modules/truffle/build/cli.bundled.js:318032:16
      at Promise (<anonymous>)
      at node_modules/truffle/build/cli.bundled.js:318023:18
      at <anonymous>
      at process._tickCallback (internal/process/next_tick.js:188:7) 
```
